Demo/src/main/java/com/braintreepayments/demo/MainActivity.java

```diff
@@ -242,29 +207,58 @@ protected void reset() {
 
     @Override
     protected void onAuthorizationFetched() {
-        try {
-            mBraintreeFragment = BraintreeFragment.newInstance(this, mAuthorization);
+        DropInRequest dropInRequest = new DropInRequest()
+                .clientToken(mAuthorization)
+                .requestThreeDSecureVerification(Settings.isThreeDSecureEnabled(this))
+                .collectDeviceData(Settings.shouldCollectDeviceData(this))
+                .googlePaymentRequest(getGooglePaymentRequest())
+                .maskCardNumber(true)
+                .maskSecurityCode(true)
+                .allowVaultCardOverride(Settings.isSaveCardCheckBoxVisible(this))
+                .vaultCard(Settings.defaultVaultSetting(this))
+                .vaultManager(Settings.isVaultManagerEnabled(this))
+                .cardholderNameStatus(Settings.getCardholderNameStatus(this));
 
-            if (ClientToken.fromString(mAuthorization) instanceof ClientToken) {
-                DropInResult.fetchDropInResult(this, mAuthorization, this);
-            } else {
-                mAddPaymentMethodButton.setVisibility(VISIBLE);
-            }
-        } catch (InvalidArgumentException e) {
-            showDialog(e.getMessage());
+        if (Settings.isThreeDSecureEnabled(this)) {
+            dropInRequest.threeDSecureRequest(demoThreeDSecureRequest());
         }
+
+        dropInClient = new DropInClient(this, mAuthorization, dropInRequest);
+        dropInClient.fetchMostRecentPaymentMethod(this, new FetchMostRecentPaymentMethodCallback() {
+            @Override
+            public void onResult(DropInResult dropInResult, Exception error) {
+                if (dropInResult != null) {
+                    handleDropInResult(dropInResult);
+                } else {
+                    mAddPaymentMethodButton.setVisibility(VISIBLE);
+                }
+            }
+        });
+
+        dropInClient.deliverBrowserSwitchResult(this, new DropInResultCallback() {
```

Comment:
How come we're calling `deliverBrowserSwitchResult` in `onAuthorizationFetched`? I feel like I'm missing something here -- wouldn't the authorization be fetched well before any browser switching occurs?

Reply:
Yeah the only reason is for process kill recovery. We'll probably look into caching the client token somehow, but we want to restructure the architecture a bit more first.